function-server/main.py

```diff
@@ -47,10 +52,19 @@ def process_msg(msg, handle):
     try:
         sys.stderr = stderr
         sys.stdout = stdout
-        r = handle(msg['context'], msg['payload'])
+        with ThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(handle, msg['context'], msg['payload'])
+            timeout = msg['context'].get('timeout', None)
+            r = future.result(timeout / 1000)
```

Comment:
If timeout is not set, this will result in a `None / 1000`, which will return an error.

Reply:
derp.. last minute addition that I didn't thoroughly check.